doc/scaling_spire.md

```diff
@@ -0,0 +1,42 @@
+Managing a shared datastore across geographically dispersed locations or across cloud providers may represent a challenge. In such circumstances, a solution is to configure SPIRE servers can be configured in Nested topology.
+
+## Nested SPIRE
+
+Nested SPIRE allows SPIRE servers to be chained together, and for all servers to still issue identities in the same trust domain. It works by co-locating an agent with downstream servers. Each downstream server loads the CA credentials from the Workload API of the top or global SPIFFE implementation. The downstream node performs both node and workload attestation, which in turn allows automating the bring-up of new downstream servers. 
+ 
+This topology is well suited for multi-cloud deployments. It can be reasoned as the top server being a global server (or set of servers for high availability). The downstream servers may be deployed at either regional or cluster level. Due to the ability to mix and match node attestors, the downstream servers can reside in different clouds. 
+ 
+In this configuration, the top tier SPIRE servers hold the root certificate/key, and the downstream servers request an intermediate signing certificate to use as the server's X.509 signing authority. It provides for resilience as the top tier can go down, and intermediate certificates will continue to operate. 
+ 
+Complimentary to scaling SPIRE servers horizontally to deal with load, a nested topology helps to segment failure domains.
+
+## Federated SPIRE
+
+A requirement or objective may exist to make use of multiple roots of trust: perhaps because an organization has different organizational divisions with different administrators, or because they have separate staging and production environments that occasionally need to communicate. A similar use-case is SPIFFE interoperability between organizations, such as between a cloud provider and its customers. Both use-cases require a well-defined, interoperable method for a workload in one trust domain to authenticate a workload in a different trust domain.
+Trust between the different trust domains is established by authenticating the respective Bundle Endpoint. For additional detail on how this is achieved, refer to the following SPIFFE spec that describe the mechanism: 
+https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE_Trust_Domain_and_Bundle.md#5-spiffe-bundle-endpoint
+
+# Sizing Clusters
+
+
+This table provides a sample reference for calculating resource allocation for SPIRE servers SPIRE deployments. It displays for different combinations of "Number of Workloads" and "Number of Agents", a corresponding baseline SPIRE Server CPU/Memory configuration. These numbers are intended to serve as sample guidance, and are not representative of upper bound scale limits nor configuration maximums. The number of workloads and agents shown do not represent the theoretically possible SPIRE deployment scale.
```

Comment:
Suggest replacing:
"These numbers are intended to serve as sample guidance, and are not representative of upper bound scale limits nor configuration maximums. The number of workloads and agents shown do not represent the theoretically possible SPIRE deployment scale."

With something stronger, like:

These reference numbers are based on experiments conducted in a testing environment.  They are meant as order-of-magnitude guidelines only, and do not represent a performance guarantee for any particular user environment.   Note that datastore performance is not addressed here, and can potentially limit SPIRE performance.  There are many factors that will influence SPIRE performance including, but not limited to:

- SVID and root certificate TTLs
- Number and distribution of workloads per node
- Heavy JWT-SVID use (because JWTs must be signed as needed, rather than pre-stashed like x509s)
- Frequency of registration changes
- Other processes running on a SPIRE Server node

Reply:
Replaced with the suggestion.